### PR TITLE
Fixes Version Selector UI defect MAGN-7250

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
### Purpose

This PR fixes https://github.com/DynamoDS/Dynamo/issues/4194, where Dynamo version selector and visual programming panel at far apart.

- When a user selects the specific Dynamo version using version selector, we no longer need the version selector UI in the current session of Revit, and hence it was disabled before. 

- This PR now hides the Ribbon Panel for the version selector so that we don't show two Dynamo icon on Ribbon.

- For Dynamo 0.8.2 onwards, the Dynamo Versions panel is renamed to "Visual Programming" so we show the Dynamo button at the same location. 

- We can't do as above for the older version of Dynamo because the previous version of Dynamo always creates a new panel named with "Visual Programming". And if this panel already exists, it throws an error. 0.8.2 onwards DynamoRevitApp checks for any available panel with this name and tries to add the Dynamo button to existing panel.

#### When Revit is launched:
image

#### If Dynamo 0.8.2 is launched 
image

#### If pre 0.8.2 Dynamo is launched
image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@Randy-Ma

### CODE REVIEW

File: src/DynamoRevit/DynamoRevitApp.cs [hunk 1]
```diff
@@ -53,8 +53,10 @@ public Result OnStartup(UIControlledApplication application)
                 ElementBinder.IsEnabled = true;
 
                 // Create new ribbon panel
-                RibbonPanel ribbonPanel =
-                    application.CreateRibbonPanel(Resources.App_Description);
+                var panels = application.GetRibbonPanels();
+                var ribbonPanel = panels.FirstOrDefault(p => p.Name.Contains(Resources.App_Description));
+                if(null == ribbonPanel)
+                    ribbonPanel = application.CreateRibbonPanel(Resources.App_Description);
```
Comment (Randy-Ma) [inline src/DynamoRevit/DynamoRevitApp.cs]: Shall we make the same check for creating the panel in DynamoRevit?

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs [hunk 2]
```diff
@@ -124,24 +121,52 @@ public static bool LaunchDynamoCommand(int index, ExternalCommandData commandDat
...
+                    .Invoke(revitCmd, new object[] { commandData, message, null });
+
+                uiApplication = null; //release application, no more needed.
+                return true;
+            }
+            catch (Exception)
+            {
+                splitButton.Enabled = true; //Disable the split button, no more needed.
+                splitButton.Visible = true; //Hide it from the UI
+                ribbonPanel.Visible = true; //Hide the panel
```
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: It looks like the comments are not correct for the three lines.

File: src/Legacy/DynamoRevitVersionSelector/Properties/Resources.en-US.resx [hunk 3]
```diff
@@ -133,4 +133,7 @@
   <data name="DynamoVersionTooltip" xml:space="preserve">
     <value>Launches Dynamo {0}, to launch a different version of Dynamo you may need to restart Revit</value>
   </data>
+  <data name="VisualProgramming" xml:space="preserve">
```
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/Properties/Resources.en-US.resx]: Shall we use the existing resource item named as App_Description?

Comment (sharadkjaiswal) [inline src/Legacy/DynamoRevitVersionSelector/Properties/Resources.en-US.resx]: App_Description is defined in DynamoRevit project, and that dll doesn't get loaded until DynamoRevitVersionSelector is loaded. Also, DynamoRevit shouldn't depend on version selector, so there we can't used resource defined here.

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs (hunk 2 above)
Comment (sharadkjaiswal) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: good catch, let me correct it.

**Comment by Randy-Ma** [comment]
LGTM

**Pull request merged**